src/Popper.jsx

```diff
@@ -38,15 +39,22 @@ class Popper extends Component {
     }
   }
 
-  componentDidUpdate(lastProps) {
+  componentDidUpdate(lastProps, lastState) {
+    const { children, eventsEnabled, onFlip, placement } = this.props
+    const { data: { placement: currentPlacement, flipped } } = this.state
+
+    if (lastState.data && lastState.data.flipped !== flipped) {
+      onFlip && onFlip({ placement: currentPlacement, flipped })
```

Comment:
should move the `onFlip &&` check to the if condition: `onFlip && lastState.data && lastState.data.flipped !== flipped`